### ISSUE TITLE
scm/resourceState/context command parameter sometimes undefined when list is updating from a change

### ISSUE THREAD

**Issue by RandomEngy** [microsoft/vscode#134333]
Does this issue occur when all extensions are disabled?: Yes (All except for my extension)

- VS Code Version: 1.60.2
- OS Version: Windows 10 20H2 19042.1237

Steps to Reproduce:

1. Download code at https://github.com/RandomEngy/VSCodeGitDiffAndMergeTool - This is a 1-file small extension. It handles context menu actions for files with merge conflicts.
2. Register a merge tool with git as described in https://github.com/RandomEngy/VSCodeGitDiffAndMergeTool
3. Open a large repository (1000s of files) and create several merge conflicts.
4. Open the Source Control tab, right click on an item with a merge conflict and click "Launch merge tool"
5. Quickly close the merge editor (do not have to accept the merge resolution, makes for easier testing)
6. Right click on another item and click "Launch merge tool" while the "Merge changes" list is refreshing from the previous resolution
7. Repeat step 6 until you see the error message "Could not launch merge tool. VSCode did not supply command parameters.  Try again in a few seconds." This usually happens about one in every 6 times for me, but you might need to try a dozen times or so.

The parameter passed to my command handler function is undefined. I think the repro is easier with a larger repository because it takes longer to do the list refresh.

It is registered like this:

	"contributes": {
		"commands": [
			{
				"command": "gitdiffandmergetool.merge",
				"title": "Launch merge tool"
			}
		],
		"menus": {
			"scm/resourceState/context": [
				{
					"command": "gitdiffandmergetool.merge",
					"when": "scmProvider == git && scmResourceGroup == merge"
				},

I run into this quite frequently when resolving merge conflicts, since you want to open the next conflict as soon as possible after resolving the previous.

**Comment by eamodio** [comment]
Yeah, I keep seeing this often too. It doesn't always seem to happen on refresh, but that is likely what is causing it. It also has seemly gotten much worse over the last couple VS Code releases.

**Comment by eamodio** [comment]
@lszomoru FYI, its pretty easy to reproduce this issue because of the refresh every 5s bug.

**Comment by lszomoru** [comment]
@eamodio, are you still able to reproduce with the latest version of VS Code?
I am asking as the SCM tree has been completed refactored and I am curious of that helped. Thanks!

**Comment by eamodio** [comment]
@lszomoru I haven't noticed this behavior in a bit -- so I think it is fixed.

**Comment by lszomoru** [comment]
@eamodio, thanks for the update. Let's close this and will reopen in case you are able to reproduce again.
